client/my-sites/theme/main.jsx

```diff
@@ -609,8 +622,12 @@ export default connect(
 		const backPath = getBackPath( state );
 		const currentUserId = getCurrentUserId( state );
 		const isCurrentUserPaid = isUserPaid( state, currentUserId );
-		const theme = getTheme( state, siteIdOrWpcom, id );
+		// Fallback to 'wpcom' source for wpcom themes on Jetpack target sites
+		const theme = getTheme( state, siteIdOrWpcom, id ) || getTheme( state, 'wpcom', id );
 		const error = theme ? false : getThemeRequestErrors( state, id, siteIdOrWpcom );
+		const isWpcomTheme = theme && theme.screenshots;
```

Comment:
Hmm, we have that [util](https://github.com/Automattic/wp-calypso/blob/1e5232b0e3da31ebd39e59eeedebea743e57efda/client/state/themes/utils.js#L194-L203) -- which due to the check it performs won't work here. But we should really try to unify how we do this, instead of having different ways checking for different things.

The util is used to filter wpcom themes from the JP "Uploaded themes" list. I wonder if we should turn it into a selector and just check for the  `screenshots` field (or whatever) instead.